plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusClient.java

```diff
@@ -180,7 +180,7 @@ private Optional<String> getBearerAuthInfoFromFile(Optional<File> bearerTokenFil
     {
         return bearerTokenFile.map(tokenFileName -> {
             try {
-                return readString(tokenFileName.toPath(), UTF_8);
+                return readString(tokenFileName.toPath(), UTF_8).replace("\n", "").replace("\r", "");
```

Comment:
I can't reproduce the issue locally. How did you confirm characters at the end of the file?

Reply:
I think the token file was likely created on Windows hence a different line terminator.

Reply:
Should this PR be merged or is this approach wrong since the same issue might apply in other scenarios and we need a more centralized fix @ebyhr @hashhar ?

Reply:
I think the approach is wrong. It's user responsibility to make sure the token file is created properly.

The most we can do would be to strip any trailing newlines (not ALL newlines like it currently does) which is safe.

Reply:
@jasonrubenstein If you still have the token file can you run `xxd <token_file>` (if you have WSL) or `Format-Hex <token-file>` in PowerShell if you don't have WSL.

If the issue is a trailing newline we should fix that but replacing all newlines is a step too far.

Reply:
we verified that the source file did not have any new line characters at the end of the token or the end of the file. The characters were present at the result of the `readString`.

Reply:
`readString` retains newlines from the source file. So if the file has a trailing newline that would explain what you saw.

I could not reproduce `readString` adding additional newlines in this code with a file that didn't contain newlines.

Reply:
so being able to confirm that would be useful. And then we can modify code to strip ONLY trailing newlines.

Reply:
The token was created in Ubuntu (not WSL; Ubuntu AMI on an EC2 instance), and Trino is running on jdk-22-22.0.2-9

I can't share the file yet (We'll create a new one with an unusable token to try a repro case), but we confirmed that the file terminates with an EOF and not any CR or LF.

Reply:
[dummy_token.txt](https://github.com/user-attachments/files/17311687/dummy_token.txt)
Note that the file we use does not have a .txt extension, but GitHub comments here refused the file without a valid extension. 

Screenshot from vscode, 1 line, no CR/LF or LF, no other control characters
![image](https://github.com/user-attachments/assets/36c741dd-4aef-4dcb-9f8e-9b38afbd2f87)

Reply:
Cannot reproduce:

```
jshell> var f = new File("/Users/ashhar.hasan/Downloads/dummy_token.txt");
f ==> /Users/ashhar.hasan/Downloads/dummy_token.txt

jshell> var x = Files.readString(f.toPath())
x ==> "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ik1H ... uaXV8-oHN_4H1q8BvkLn9rZWg"

jshell> System.out.println(x)
eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ik1HTHFqOThWTkxvWGFGZnBKQ0JwZ0I0SmFLcyIsImtpZCI6Ik1HTHFqOThWTkxvWGFGZnBKQ0JwZ0I0SmFLcyJ9.eyJhdWQiOiI1ZGJkNTdjYy00MWIyLTQzMjUtODRhZi02MDMwMmNiNzgzZTkiLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC9mYTdiMWI1YS03YjM0LTQzODctOTRhZS1kMmMxNzhkZWNlZTEvIiwiaWF0IjoxNzIxODM2NTYxLCJuYmYiOjE3MjE4MzY1NjEsImV4cCI6MTcyMTg0MDQ2MSwiYWlvIjoiRTJkZ1lLaUtmL3ppbytqU2lYMWk5OU4wM2lTYUF3QT0iLCJhcHBpZCI6IjBlYmE3YTM1LWM1MmYtNGQwNy1hNTBjLTM5NTllYThmYjgyMyIsImFwcGlkYWNyIjoiMSIsImlkcCI6Imh0dHBzOi8vc3RzLndpbmRvd3MubmV0L2ZhN2IxYjVhLTdiMzQtNDM4Ny05NGFlLWQyYzE3OGRlY2VlMS8iLCJvaWQiOiI0MjQxYjAwYS05MDdkLTQ1YTctYTJlZC1iZTcyZjM0YmYyY2QiLCJyaCI6IjAuQVNZQVdodDctalI3aDBPVXJ0TEJlTjdPNGN4WHZWMnlRU1ZEaEs5Z01DeTNnLWttQUFBLiIsInN1YiI6IjQyNDFiMDBhLTkwN2QtNDVhNy1hMmVkLWJlNzJmMzRiZjJjZCIsInRpZCI6ImZhN2IxYjVhLTdiMzQtNDM4Ny05NGFlLWQyYzE3OGRlY2VlMSIsInV0aSI6IlVidnNYTWtHYWt1MGFyOEVkVmFtQUEiLCJ2ZXIiOiIxLjAifQ.CZoQV-8-OsujiFuQgDmqcz25RPAWTx-0LmAJg87Z4QY_P_YWDkEJo-RX_ggHGZXh10aWUWy1OIlpfLSh5OhStxjXdFLjfXwUAEpfpn15_Jp-dG9mR6KnMpEb7FG8u7l43MmfqyJCTVmMjTBdvAfR51F1HjHQIMoIYJz8-MzSfRNC3F40w7jjL_HXbYSd66pfUpe8QSteDCWz00p3SzuEpiBhSeq4LAmTYXAO6w5a6E99ZW2eYhNP1Av1i9d1QnI-fMX3cgdcNDX3Y5nGrAequiY_W-BRF0ADopHolNMHwsyZWARPAI9wUQue1cx-cuaXV8-oHN_4H1q8BvkLn9rZWg

jshell> var y = x.replaceAll("\n", "").replaceAll("\r", "");
y ==> "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ik1H ... uaXV8-oHN_4H1q8BvkLn9rZWg"

jshell> x.equals(y)
$5 ==> true
```

Does changing the patch to just the below solve the issue for you locally:

```java
return readString(tokenFileName.toPath(), UTF_8).stripTrailing()
```

Also can you log the bytes Trino sees in your case with a local patch?